src/System Application/App/Extension Management/src/ExtensionDeploymentStatus.Page.al

```diff
@@ -99,6 +100,49 @@ page 2508 "Extension Deployment Status"
                     ExtnDeploymentStatusDetail.Run();
                 end;
             }
+            action(Refresh)
+            {
+                ApplicationArea = All;
+                ToolTip = 'Refresh the deployment details.';
+                Image = Refresh;
+
+                trigger OnAction()
+                var
+                    NavAppTenantOperationTable: Record "NAV App Tenant Operation";
+                    ExtensionOperationImpl: Codeunit "Extension Operation Impl";
+                begin
+                    NavAppTenantOperationTable.CopyFilters(Rec);
+                    if NavAppTenantOperationTable.FindSet() then
+                        repeat
+                            ExtensionOperationImpl.RefreshStatus(NavAppTenantOperationTable."Operation ID");
```

Comment:
We only need to refresh status if the status is `Unknown`, `NotFound`, or `InProgress`. Once it's `Failed` or `Completed` it's not going to change anymore. This should cut down the number of operations that actually need refreshing. Might be easier still to just set a filter on `NavAppTenantOperationTable` before `FindSet` to get the records that aren't in `Failed` or `Completed` state.